src/TRestRawSignalRecoverChannelsProcess.cxx

```diff
@@ -170,48 +178,150 @@ TRestEvent* TRestRawSignalRecoverChannelsProcess::ProcessEvent(TRestEvent* evInp
 
     Int_t idL;
     Int_t idR;
+    Int_t idL2;
+    Int_t idR2;
     for (unsigned int x = 0; x < fChannelIds.size(); x++) {
-        GetAdjacentSignalIds(fChannelIds[x], idL, idR);
-        // cout << "Channel id : " << fChannelIds[x] << " Left : " << idL << " Right
-        // : " << idR << endl;
+        GetAdjacentSignalIds(fChannelIds[x], idL, idR, idL2, idR2);
+        // cout << "Channel id : " << fChannelIds[x] << " Left : " << idL << " Right : " << idR << endl;
 
         if (fOutputSignalEvent->GetSignalIndex(fChannelIds[x]) > 0)
             fOutputSignalEvent->RemoveSignalWithId(fChannelIds[x]);
 
-        if (idL == -1 || idR == -1) continue;
+        if (idL == -1 || idR == -1 || idL2 == -1 || idR2 == -1) continue;
```

Comment:
Finally idL2 and idR2 are only used here. It means that if the second adjacent channel is not present it will not do the correction.